libraries/Microsoft.Bot.Builder.Dialogs/Recognizer.cs

```diff
@@ -70,22 +70,39 @@ public virtual async Task<T> RecognizeAsync<T>(DialogContext dialogContext, Acti
             return result;
         }
 
-        protected virtual Dictionary<string, string> FillRecognizerResultTelemetryProperties(RecognizerResult recognizerResult, Dictionary<string, string> telemetryProperties, ITurnContext turnContext = null)
+        protected virtual Dictionary<string, string> FillRecognizerResultTelemetryProperties(RecognizerResult recognizerResult, Dictionary<string, string> telemetryProperties, DialogContext dialogContext = null)
         {
```

Comment:
Was this shipped in 4.8?  I would think this is a breaking change to change this signature.

Reply:
It was. I had misunderstood the way the expression value worked, which means that the current implementation doesn't get the right application ID, so I think I do need the dialog context unless there is another way to get the value -> https://github.com/microsoft/botbuilder-dotnet/blob/0c8597686b81e035d0590687050c3ee8481ed2f3/libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/LUIS/LuisAdaptiveRecognizer.cs#L140-L156  

If I do need DialogContext, given that adaptive isn't GA yet and we have not documented any of the telemetry bits yet, I think this is an acceptable change for us to make here. Thoughts?